units/host_drawdown.go

```diff
@@ -137,6 +138,17 @@ func (j *hostDrawdownJob) checkAndDecommission(ctx context.Context, h *host.Host
 		idleThreshold = idleTaskGroupDrawdownCutoff
 	}
 
+	if !h.LastTaskCompletedTime.IsZero() {
+		taskQueue, err := model.FindDistroTaskQueue(ctx, h.Distro.Id)
```

Comment:
Since this job already runs for a single distro per job, we could reduce the number of repeated find on the same task queue by just looking up the distro's task queue once before checking the hosts.